packages/renderer/src/lib/preferences/ExtensionIcon.svelte

```diff
@@ -0,0 +1,26 @@
+<script lang="ts">
+import Fa from 'svelte-fa';
+import type { ExtensionInfo } from '../../../../main/src/plugin/api/extension-info';
+import { faPuzzlePiece } from '@fortawesome/free-solid-svg-icons';
+
+export let extension: ExtensionInfo;
+
+let icon: string | undefined = undefined;
+$: {
+  if (extension.icon) {
+    if (typeof extension.icon === 'string') {
+      icon = extension.icon;
+    } else if (extension.icon.dark) {
+      // for now use dark theme
+      icon = extension.icon.dark;
+    }
```

Comment:
as we'll introduce light theme soon, I would think that it has the logic for light and dark

Reply:
Agreed, but I think the right solution is that we need an appearance-util.js that'll tell you if you should use light or dark, maybe with a utility to pull the right thing from `string | { light: string; dark: string }`, then used here + ProviderLogo (maybe others?). Are you ok doing that as follow-up PR?

Reply:
Ok, I caved and threw my thoughts in PR #5117 and will wait until tomorrow for initial feedback. If that approach makes sense I can wait and rebase this; otherwise I would prefer to merge and use #5116 to solve this along with the other places we already have this problem.

Reply:
can be follow-up